addons/mail/static/src/web/thread_service_patch.js

```diff
@@ -149,8 +149,11 @@ patch(ThreadService.prototype, "mail/web", {
     async insertSuggestedRecipients(thread, dataList) {
         const recipients = [];
         for (const data of dataList) {
-            const [partner_id, emailInfo, lang, reason] = data;
-            const [name, email] = emailInfo && parseEmail(emailInfo);
+            const [partner_id, emailInfo, lang, reason, defaultCreateValues] = data;
+            let [name, email] = emailInfo && parseEmail(emailInfo);
+            if (name === email && defaultCreateValues && defaultCreateValues.name) {
+                name = defaultCreateValues.name;
+            }
```

Comment:
I would not name extra param `defaultCreateValues`: this makes this part of data too black-boxed, so there's no idea what it actually contains.

From what I see, it contains customer info, so I would simply name it `customerInfo`

Reply:
@alexkuhn Thanks for the review! Updated :+1: